src/image/jpeg_io.cpp

```diff
@@ -50,6 +50,10 @@ void parse_jpeg(const char* data, size_t length,
     jpeg_mem_src(&cinfo, (unsigned char*)data, length); // Specify data source for decompression
     jpeg_read_header(&cinfo, TRUE); // Read file header, set default decompression parameters
 
+    if (cinfo.out_color_space != JCS_GRAYSCALE && cinfo.out_color_space != JCS_RGB) {
+      log_and_throw(std::string("Unsupported colorspace format. Currently, only RGB and Grayscale are supported."));
```

Comment:
How does this interact with the `ignore_failure` parameter of `load_images` (in `src/unity/lib/image_util.cpp`?

Reply:
If ignore_failure parameter is set to true then it logs the error above along with the url of the offending file.

image_util.cpp Line [188:](https://github.com/apple/turicreate/blob/af42916ca1b2816843c56b749e4166e6a6d709b6/src/unity/lib/image_util.cpp#L188)  logprogress_stream << error << "\t" << " file: " << sanitize_url(*iter) << std::endl;

which logs the error as:

"Unsupported colorspace format. Currently, only RGB and Grayscale are supported.	 file: ./images/sample.jpg"

Reply:
Ah, got it, I see now. Thanks for the explanation!